packages/form-control/src/hooks/useAccessibleField.ts

```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react';
+
+import { useIdAllocator } from '@leafygreen-ui/hooks';
+
+import { AccessibleFieldProps, LabelProps } from './types';
+
+export const useLabel = ({
+  label,
+  id: idProp,
+  'aria-labelledby': ariaLabelledby,
+  'aria-label': ariaLabel,
+}: LabelProps) => {
+  const inputId = useIdAllocator({ prefix: 'lg-input', id: idProp });
+  const labelId = useIdAllocator({ prefix: 'lg-label' });
+
+  const labelProps = {
+    id: labelId,
+    htmlFor: inputId,
+  };
+
+  if (!label && !ariaLabelledby && !ariaLabel) {
+    console.warn(
+      'For screen-reader accessibility, label, aria-labelledby, or aria-label must be provided',
+    );
+  }
+
+  const fieldProps = {
+    id: inputId,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby
+      ? `${ariaLabelledby} ${labelId}`
+      : labelId,
+  };
+
+  return {
+    labelProps,
+    fieldProps,
+  };
+};
+
+export const useAccessibleField = ({
+  label,
+  id,
+  'aria-labelledby': ariaLabelledby,
+  'aria-label': ariaLabel,
+  description,
+  errorMessage,
+  validationState,
+}: AccessibleFieldProps) => {
+  const { labelProps, fieldProps } = useLabel({
+    label,
+    id,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby,
+  });
+
+  const descriptionId = useIdAllocator({ prefix: 'lg-description' });
+  const errorMessageId = useIdAllocator({ prefix: 'lg-errorMessage' });
```

Comment:
I don't think we are addressing if a custom message container is used, so if the consumer is passing in `aria-describedby`.

Reply:
ok i can spread that, great call